File: verify_images.py
import os
import re
import requests
from concurrent.futures import ThreadPoolExecutor

def get_image_urls(directory):
    urls = set()
    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith('.html'):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8') as f:
                    content = f.read()
                    # Find src="..."
                    matches = re.findall(r'src=["\'](http[^"\']+)["\']', content)
                    urls.update(matches)
                    # Find poster="..."
                    matches = re.findall(r'poster=["\'](http[^"\']+)["\']', content)
                    urls.update(matches)
    return list(urls)

def check_url(url):
    try:
        response = requests.head(url, timeout=5)
        if response.status_code != 200:
            return url, response.status_code
    except Exception as e:
        return url, str(e)
    return url, 200

def main():
    directory = "."
    print(f"Scanning {directory} for HTML files...")
    urls = get_image_urls(directory)
    print(f"Found {len(urls)} unique image URLs.")
    
    broken_urls = []
    with ThreadPoolExecutor(max_workers=10) as executor:
        results = executor.map(check_url, urls)
        for url, status in results:
            if status != 200:
                print(f"BROKEN: {url} (Status: {status})")
                broken_urls.append(url)
            else:
                pass # print(f"OK: {url}")

    if not broken_urls:
        print("All image URLs are valid!")
    else:
        print(f"Found {len(broken_urls)} broken URLs.")

if __name__ == "__main__":
    main()

File: verify_images.js
const fs = require('fs');
const path = require('path');
const https = require('https');
const http = require('http');

function getImageUrls(dir) {
    const files = fs.readdirSync(dir);
    let urls = new Set();
    
    files.forEach(file => {
        if (file.endsWith('.html')) {
            const content = fs.readFileSync(path.join(dir, file), 'utf8');
            const regex = /src=["'](https?:\/\/[^"']+)["']/g;
            let match;
            while ((match = regex.exec(content)) !== null) {
                urls.add(match[1]);
            }
            const regexPoster = /poster=["'](https?:\/\/[^"']+)["']/g;
            while ((match = regexPoster.exec(content)) !== null) {
                urls.add(match[1]);
            }
        }
    });
    return Array.from(urls);
}

function checkUrl(url) {
    return new Promise((resolve) => {
        const protocol = url.startsWith('https') ? https : http;
        const req = protocol.request(url, { method: 'HEAD' }, (res) => {
            if (res.statusCode === 200 || res.statusCode === 301 || res.statusCode === 302) {
                resolve({ url, status: res.statusCode });
            } else {
                resolve({ url, status: res.statusCode, broken: true });
            }
        });
        req.on('error', (e) => {
            resolve({ url, error: e.message, broken: true });
        });
        req.end();
    });
}

async function main() {
    console.log("Scanning directory...");
    const urls = getImageUrls('.');
    console.log(`Found ${urls.length} unique URLs.`);
    
    const results = await Promise.all(urls.map(checkUrl));
    const broken = results.filter(r => r.broken);
    
    if (broken.length === 0) {
        console.log("All image URLs are valid!");
    } else {
        console.log(`Found ${broken.length} broken URLs:`);
        broken.forEach(b => console.log(`${b.url} (Status: ${b.status || b.error})`));
    }
}

main();
